### PR TITLE
`rr.serve` -> `rr.serve_web`, `rr.connect` -> `rr.connect_tcp`

### LINKED ISSUE #7766: `rr.serve` should be `rr.serve_websocket`, `rr.connect` should be `rr.connect_tcp`
And similarly, `rerun --serve` should be `rerun --serve-websocket` (or something along these lines, you get the idea).

Users get understandably confused by this. They expect a serving process (`rr.serve()`) to be ready to accept connections from other processes (`rr.connect()`), except the two things are in reality completely unrelated: one is hosting a websocket server to be polled by the web-viewer, while the other is trying to connect to the TCP sdk comms pipeline. 

See e.g. this thread: https://discord.com/channels/1062300748202921994/1295914647165210634

### PR DESCRIPTION
**Pull request by teh-cmc**
Title.

* Fixes #7766 

* [x] :sheep:

### CODE REVIEW

**Comment by teh-cmc** [comment]
@rerun-bot full-check

File: rerun_py/rerun_sdk/rerun/sinks.py [hunk 1]
```diff
@@ -25,6 +26,55 @@ def connect(
     flush_timeout_sec: float | None = 2.0,
     default_blueprint: BlueprintLike | None = None,
     recording: RecordingStream | None = None,
+) -> None:
```
Comment (jleibs) [inline rerun_py/rerun_sdk/rerun/sinks.py:29]: Please add `@deprecated` decorator as in:
https://github.com/rerun-io/rerun/pull/3564

**Review by jleibs — APPROVED** [review]

**Comment by teh-cmc** [comment]
I'm not quite sure what these errors mean:
```
mkdocs_autorefs: common/initialization_functions.md: from /home/runner/work/rerun/rerun/rerun_py/rerun_sdk/rerun/__init__.py:0: (rerun) Could not find cross-reference target 'rerun.connect_tcp'
mkdocs_autorefs: common/initialization_functions.md: from /home/runner/work/rerun/rerun/rerun_py/rerun_sdk/rerun/__init__.py:0: (rerun) Could not find cross-reference target 'rerun.serve_web'
```
and I'm not able to build the docs locally:
* #7913

File: crates/top/re_sdk/src/recording_stream.rs [hunk 2]
```diff
@@ -503,13 +542,62 @@ impl RecordingStreamBuilder {
...
+    /// The WebSocket server will buffer all log data in memory so that late connecting viewers will get all the data.
+    /// You can limit the amount of data buffered by the WebSocket server with the `server_memory_limit` argument.
+    /// Once reached, the earliest logged data will be dropped.
+    /// Note that this means that static data may be dropped if logged early (see <https://github.com/rerun-io/rerun/issues/5531>).
+    ///
+    /// ## Example
+    ///
+    /// ```ignore
+    /// let rec = re_sdk::RecordingStreamBuilder::new("rerun_example_app")
+    ///     .serve("0.0.0.0",
```
Comment (zehiko) [inline crates/top/re_sdk/src/recording_stream.rs:584]: nit: ``serve_web``

**Review by zehiko — APPROVED** [review]

**Comment by teh-cmc** [comment]
@rerun-bot full-check

**Pull request merged**